docs/git-resolver.md

```diff
@@ -75,7 +75,9 @@ The differences between the two modes are:
 
 ### Git Clone with git clone
 
-Git clone with `git clone` is supported for anonymous and authenticated cloning.
+Git clone with `git clone` is supported for anonymous and authenticated cloning
+This mode shallow-clones the git repo before shallow-fetching and checking
```

Comment:
that's a bit too low level git plumbing to understand?

Reply:
Yeah, that's fair

Reply:
I removed the bit about `shallow` fetching, but left in the steps `shallow-clone, fetch, then checkout`. I added a note detailing when users may need to care about the git plumbing, since there is that possibility. Does that read better now?